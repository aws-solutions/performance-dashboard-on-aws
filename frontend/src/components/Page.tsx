/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React, { useEffect } from "react";
import { Route, RouteProps } from "react-router-dom";
import { useSettings } from "../hooks";
import { useTranslation } from "react-i18next";

interface Props extends RouteProps {
    title: string;
}

function Page({ title, ...routeProps }: Props) {
    const { t } = useTranslation();
    const { settings } = useSettings();

    useEffect(() => {
        const appTitle =
            settings.navbarTitle || t("PageTitle.Default") || "Performance Dashboard on AWS";

        document.title = title ? `${title} - ${appTitle}` : appTitle;
    }, [settings.navbarTitle, t, title]);

    return <Route {...routeProps} />;
}

export default Page;
