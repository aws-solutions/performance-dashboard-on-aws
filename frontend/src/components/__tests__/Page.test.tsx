/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import Page from "../Page";

jest.mock("../../hooks");
jest.mock("../../services/BackendService");

describe("Page", () => {
    const history = createMemoryHistory();

    test("renders a page", async () => {
        const { baseElement } = render(
            <Router history={history}>
                <Page title="this is a title" />
            </Router>,
        );
        expect(baseElement).toMatchSnapshot();
    });

    test("should set document title", async () => {
        render(
            <Router history={history}>
                <Page title="this is a title" />
            </Router>,
        );
        expect(document.title).toBe("this is a title - Performance Dashboard");
    });
});
