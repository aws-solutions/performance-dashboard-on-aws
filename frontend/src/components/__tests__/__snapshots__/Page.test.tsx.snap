// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders a page 1`] = `
<body>
  <div />
</body>
`;
